src/GovUk.Education.ExploreEducationStatistics.Admin/Services/DataSetUploadRepository.cs

```diff
@@ -0,0 +1,58 @@
+#nullable enable
+using AutoMapper;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Admin.ViewModels;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using static GovUk.Education.ExploreEducationStatistics.Common.BlobContainers;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services;
+
+public class DataSetUploadRepository(
+    ContentDbContext contentDbContext,
+    IPrivateBlobStorageService privateBlobStorageService,
+    IMapper mapper) : IDataSetUploadRepository
+{
+    public async Task<Either<ActionResult, List<DataSetUploadViewModel>>> ListAll(
+        Guid releaseVersionId,
+        CancellationToken cancellationToken = default)
+    {
+        return await contentDbContext.DataSetUploads
+            .Where(uploads => uploads.ReleaseVersionId == releaseVersionId)
+            .Select(entity => mapper.Map<DataSetUploadViewModel>(entity))
+            .ToListAsync(cancellationToken);
+    }
+
+    public async Task<Either<ActionResult, Unit>> Delete(
+        Guid releaseVersionId,
+        Guid dataSetUploadId,
+        CancellationToken cancellationToken = default)
+    {
+        return await contentDbContext.DataSetUploads
+            .SingleOrNotFoundAsync(upload =>
+                upload.ReleaseVersionId == releaseVersionId &&
+                upload.Id == dataSetUploadId,
+                cancellationToken)
+            .OnSuccess(async dataSetUpload =>
```

Comment:
Could also slightly simplify with:

```
            .OnSuccessDo(async dataSetUpload =>
            {
                contentDbContext.Remove(dataSetUpload);
                await contentDbContext.SaveChangesAsync(cancellationToken);
            })
```

It's just perform the action and then the original `dataSetUpload` parameter will be passed to the next call in the chain.